features/FxA-45-device-registration-api/README.md

```diff
@@ -0,0 +1,386 @@
+  * [How are unrecognised devices handled?](#how-are-unrecognised-devices-handled)
+  * [What is a disconnected device?](#what-is-a-disconnected-device)
+
+## Stories
+
+* As an API consumer,
+  I want to create an account with the auth server
+  using new device details
+  and receive the newly-registered device id in the response.
+
+* As an API consumer,
+  I want to create an account with the auth server
+  using a registered device id.
+
+* As an API consumer,
+  I want to log in to an auth server account
+  using new device details
+  and receive the newly-registered device id in the response.
+
+* As an API consumer,
+  I want to log in to an auth server account
+  using a registered device id.
+
+* As a logged-in API consumer,
+  I want to register new device details
+  and receive the newly-registered device id in the response.
+
+* As a logged-in API consumer,
+  I want to update the details for a registered device.
+
+* As a logged-in API consumer,
+  I want to get a list of all the registered device details.
+
+* As a logged-in API consumer,
+  I want to deregister a registered device
+  and remove all of its details from the server.
+
+* As a logged-in API consumer,
+  I want to receive a `"new sync device"` push notification
+  when a new device is registered.
+
+* As a logged-in API consumer,
+  I want to receive a `"device name changed"` push notification
+  when a registered device is renamed.
+
+* As a logged-in API consumer,
+  I want to receive a `"password changed"` push notification
+  when the password is changed.
+
+* As a logged-in API consumer,
+  I want to receive a `"password reset"` push notification
+  when the password is reset.
+
+* As a logged-in API consumer,
+  I want to receive a `"disconnected"` push notification
+  when a registered device transitions to the disconnected state.
+
+## Data
+
+The following details will be stored on the server
+for each registered device:
+
+```js
+{
+  // The user this device belongs to
+  "uid": "4c352927cd4f4a4aa03d7d1893d950b8",
+
+  // The active session for this device
+  "sessionToken": "27cd4f4a4aa03d7d186a2ec81cbf19d5c8a604713362df9ee15c4f4a4aa03d7d",
+
+  // A server-assigned identifier for this device
+  "id": "0f7aa00356e5416e82b3bef7bc409eef",
+
+  // A human identifier for this devices. Able to be updated by the user
+  "name": "My Phone",
+
+  // General type of device. Useful for device list icons and demographic metrics
+  "type": "mobile",
+
+  // Timestamp of when the device was first created, à la Date.now()
+  "createdAt": 1442785364807,
+
+  // The push notification endpoint for this device/user
+  "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef",
+
+  // The public key for encrypting messages to the push endpoint
+  "publicKey": "468601214f60f4828b6cd5d51d9d99d212e7c73657978955f0f5a5b7e2fa1370"
```

Comment:
I'm going to nitpick the names here, because it's not at all obvious that "callback" and "publicKey" are related and should be used as a pair.  Perhaps we can put "push" in the name of both somehow, to make it more obvious?